src/valid/type.rs

```diff
@@ -630,7 +630,12 @@ impl super::Validator {
                 self.require_type_capability(Capabilities::RAY_QUERY)?;
                 TypeInfo::new(TypeFlags::DATA | TypeFlags::SIZED, Alignment::ONE)
             }
-            Ti::BindingArray { .. } => TypeInfo::new(TypeFlags::empty(), Alignment::ONE),
+            Ti::BindingArray { base, size: _ } => {
+                if base >= handle {
+                    return Err(TypeError::UnresolvedBase(base));
+                }
+                self.types[base.index()].clone()
```

Comment:
The old code wasn't right either, but it at least had the merit of at least not promising things about the type that weren't true, like `TypeFlags::ARGUMENT`. Can we make this accurate?

Reply:
good point, thanks for your diligence! take a look at the new commit please